src/systems/spawning/spawn_resource_planets.rs

```diff
@@ -0,0 +1,38 @@
+use bevy::{
+    ecs::{
+        event::{EventReader, EventWriter},
+        system::Commands,
+    },
+    math::Quat,
+};
+use rand::{random, Rng};
+
+use crate::{
+    components::planet::Planet,
+    events::{spawn_planet_event::SpawnPlanetEvent, spawn_sprite_event::SpawnSpriteEvent},
+    resources::constants::PLANET_DISTANCE_FROM_SPACESTATION,
+};
+
+pub fn spawn_resource_planets(
+    mut commands: Commands,
+    mut spawn_sprite_event_writer: EventWriter<SpawnSpriteEvent>,
+    mut spawn_planet_event_reader: EventReader<SpawnPlanetEvent>,
+) {
+    for spawn_planet_event in spawn_planet_event_reader.read() {
+        let mut rng = rand::thread_rng();
+        let angle: f32 = rng.gen_range(0.0..360.0);
+        let planet = Planet::new(random());
```

Comment:
@MatthewGriffin should be able to randomly size the planets without worrying about the spawn zones provided the distance to the space station is larger than the maximum size we set. I'll make a PR for it later :smile:

Reply:
(Addressed)